ddtrace/tracer.py

```diff
@@ -299,12 +301,14 @@ def configure(
         self.writer.stop()
         if url:
             agent.verify_url(url)
+        sync_mode = self._is_agentless_environment()
```

Comment:
A quick comment above explaining why being in an agentless environment triggers synchronous flush mode would be beneficial for future readers